platform/ios/ios.xcodeproj/project.pbxproj

```diff
@@ -2958,6 +2917,28 @@
 		};
 /* End PBXResourcesBuildPhase section */
 
+/* Begin PBXShellScriptBuildPhase section */
+		072FAE4E2230A76900CC750D /* Insert Mapbox Access Token */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+				"$(TARGET_BUILD_DIR)/$(INFOPLIST_PATH)",
+			);
+			name = "Insert Mapbox Access Token";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "if [[ \"$CI\" ]]; then\n  echo \"CI environment, access token not required\"\n  exit 0\nelse\n  token_file=$SRCROOT/mapbox_access_token\n  token=\"$(cat $token_file 2> /dev/null)\" || token=\"$(cat ~/.mapbox 2> /dev/null)\"\n\n  if [ \"$token\" ]; then\n    plutil -replace MGLMapboxAccessToken -string $token \"$TARGET_BUILD_DIR/$INFOPLIST_PATH\"\n  else\n    echo 'error: Missing Mapbox access token'\n    echo \"error: Get an access token from <https://www.mapbox.com/studio/account/tokens/>, then create a new file at $token_file that contains the access token.\"\n  exit 1\nfi\n\nfi\n\n";
```

Comment:
There are some tradeoffs to directly including scripts in projects — they’re a bit difficult to read/review and can cause merge conflicts, but it’s perhaps a more intuitive editing experience when they’re in Xcode.

The alternative is to break this out into its own file, then invoke that file here in the project. Separate files are better formatted (in pull requests, at least) and able to be run independently, so I tend to prefer them.